src/apis/post-tags/services/post-tags.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+
+import { Transactional } from 'typeorm-transactional';
+
+import { CreatePostTagDto } from '@src/apis/post-tags/dto/create-post-tag.dto';
+import { PostTagDto } from '@src/apis/post-tags/dto/post-tag.dto';
+import { PostTagRepository } from '@src/apis/post-tags/repositories/post-tag.repository';
+
+@Injectable()
+export class PostTagsService {
+  constructor(private readonly postTagRepository: PostTagRepository) {}
+
+  @Transactional()
+  async create(userId: number, createPostTagDto: CreatePostTagDto) {
+    const existPostTag = await this.findOneByName(createPostTagDto.name);
+
+    if (existPostTag) {
+      return existPostTag;
+    }
+
+    const newPostTag = await this.postTagRepository.save({
+      userId,
+      ...createPostTagDto,
+    });
+
+    return new PostTagDto(newPostTag);
+  }
+
+  async findOneByName(name: string) {
+    const postTag = await this.postTagRepository.findOneBy({
+      name,
+    });
+
+    if (postTag === null) {
+      return;
+    }
```

Comment:
```suggestion
    if (!postTag) {
      return;
    }
```
이게 좀 더 깔끔할 것 같은데 어떤가요?

Reply:
제안해주신 코드는 falsy한 값에 대한 처리여서 다른 로직인거같아요.
정확히 null일때만 undefined를 리턴하는걸 의도했습니다.